addons/mail/static/src/owl/components/composer_text_input/composer_text_input.js

```diff
@@ -161,14 +274,84 @@ class ComposerTextInput extends Component {
      */
     _onKeydownTextarea(ev) {
         switch (ev.key) {
+            // UP, DOWN, TAB: prevent moving cursor if navigation in mention propositions
+            case 'ArrowUp':
+            case 'ArrowDown':
+            case 'Tab':
+                if (this.state.isMentionsOpen) {
+                    ev.preventDefault();
+                }
+                break;
+            // ENTER: submit the message only if the dropdown mention proposition is not displayed
             case 'Enter':
-                this._onKeydownTextareaEnter(ev);
+                if (this.state.isMentionsOpen) {
+                    ev.preventDefault();
+                } else {
+                    this._onKeydownTextareaEnter(ev);
+                }
+                break;
+        }
+    }
+
+    propositionNavigation(key) {
```

Comment:
I would split this method into small functional methods, and leave the key binding only inside `_onKeyupTextarea`

Reply:
Done but may not be what was intended